File: apps/app/ios/App/DeviceHelper.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DeviceHelper, NSObject)

RCT_EXTERN_METHOD(getDeviceModel:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: apps/app/ios/App/DeviceHelper.swift
import Foundation
import React

@objc(DeviceHelper)
class DeviceHelper: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func getDeviceModel(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    #if targetEnvironment(simulator)
      let model = "iOS Simulator (\(UIDevice.current.model))"
    #else
      let model = UIDevice.current.name
    #endif
    resolve(model)
  }
}
